overlord/snapstate/autorefresh.go

```diff
@@ -58,6 +58,30 @@ var (
 // refreshRetryDelay specified the minimum time to retry failed refreshes
 var refreshRetryDelay = 20 * time.Minute
 
+// refreshCandidate carries information about a single snap to update as part
+// of auto-refresh.
+type refreshCandidate struct {
+	SnapSetup
+}
+
+func (rh *refreshCandidate) Type() snap.Type {
+	return rh.SnapSetup.Type
+}
+
+func (rh *refreshCandidate) Base() string {
+	return rh.SnapSetup.Base
+}
+
+func (rh *refreshCandidate) MakeSnapSetup(st *state.State, snapst *SnapState, revnoOpts *RevisionOptions, snapUserID int) (*SnapSetup, error) {
```

Comment:
should snapUserID be used?

Reply:
Not really, but also I'm not sure about the final signature of this method. The original idea was to have common interface for refreshCandidate and "manually" refreshed snaps so that doUpdate logic works on both (this goes back to the bigger "draft" PR). Depending on the final structure of UpdateMany / doUpdate (and potentially having separate function(s) for autorefresh as you hinted before), MakeSnapSetup may not be needed after all.

Reply:
I can remove MakeSetup from this PR and re-introduce it later if needed if you think it's worth it?